index.js

```diff
@@ -1,15 +1,59 @@
-module.exports = ({
-  user = (ctx) => `You're the following permissions: ${ctx.command.userPerms.join(', ')}`,
-  my = (ctx) => `I'm missing the following permissions: ${ctx.command.myPerms.join(', ')}`
+const humanReadableStrings = {
+  createInvites: 'Create Invites',
+  kick: 'Kick Members',
+  ban: 'Ban Members',
+  administrator: 'Administrator',
+  manageChannels: 'Manage Channels',
+  manageGuild: 'Manage Server',
+  addReactions: 'Add Reactions',
+  auditLog: 'View Audit Log',
+  prioritySpeaker: 'Priority Speaker',
+  stream: 'Stream',
+  viewChannel: 'View Channel(s)',
+  sendMessages: 'Send Messages',
+  tts: 'Send Text-to-Speech Messages',
+  manageMessages: 'Manage Messages',
+  embed: 'Embed Links',
+  files: 'Attach Files',
+  readHistory: 'Read Message History',
+  mentionEveryone: 'Mention \@everyone, \@here, and All Roles',
+  externalEmojis: 'Use External Emoji',
+  viewInsights: 'View Server Invites',
+  connect: 'Connect (Voice)',
+  speak: 'Speak (Voice)',
+  mute: 'Mute (Voice)',
+  deafen: 'Deafen (Voice)',
+  move: 'Move (Voice)',
+  useVoiceActivity: 'Use Voice Activity',
+  nickname: 'Change Nickname',
+  manageNicknames: 'Manage Nicknames',
+  manageRoles: 'Manage Roles',
+  webhooks: 'Manage Webhooks',
+  emojis: 'Manage Emojis'
+}
+
+const exportFunc = module.exports = ({
+  humanReadable = humanReadableStrings,
+  my = (perms, ctx) => `I am missing the following permissions: ${perms.map(p => (typeof humanReadable === 'function' ? humanReadable(ctx, p) : humanReadable[p]) ?? p).join(', ')}`,
+  user = (perms, ctx) => `You are missing the following permissions: ${perms.map(p => (typeof humanReadable === 'function' ? humanReadable(ctx, p) : humanReadable[p]) ?? p).join(', ')}`,
+  sendAsRoseError = true,
+  makeResponseAReply = false
 } = {}) => {
   return (ctx) => {
-    if (ctx.command.userPerms && !ctx.command.userPerms.every(x => ctx.hasPerms(x))) {
-      throw new Error(user(ctx))
+    if (ctx.command.hasOwnProperty('myPerms') && !ctx.command.myPerms.every(x => ctx.myPerms(x))) {
+      const perms = ctx.command.myPerms.filter(p => !ctx.myPerms(p))
+      if (sendAsRoseError) throw new Error(my(perms, ctx))
```

Comment:
It might be better to use `return ctx.error()` rather than `throw new Error()`, but that is probably a useless change that has no real effect. Also, if the bot is missing the `sendMessages` permission or the replied to message was deleted, an error would be thrown when the response is sent. This realistically is not an issue but it might be better to catch it regardless. This is up to @jpbberry if he wants this to be changed as it should still be handled by the command handler.

Reply:
I'll make the transition to ctx.error() with all the middlewares together, not right now though.